pkg/apis/pipeline/v1beta1/result_defaults.go

```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import "context"
+
+// SetDefaults set the default type for TaskResult
+func (tr *TaskResult) SetDefaults(ctx context.Context) {
+	if tr != nil && tr.Type == "" {
+		// ResultsTypeString is the default value
+		tr.Type = ResultsTypeString
+	}
```

Comment:
Similar like param, this will mutate the task, if this is necessary. I will document in current `trusted task`'s design and logging fields.